frontend/src/pages/AnnouncementPage.js

```diff
@@ -0,0 +1,330 @@
+// SPDX-License-Identifier: MIT
+// SDPX-FileCopyrightText: 2021 Que Le <b.le@tu-berlin.de>
+
+import React, {
+	useEffect,
+	useState
+} from "react";
+import appwriteApi from "../api/appwriteApi";
+
+import Grid from "@mui/material/Grid";
+import { makeStyles } from "@mui/styles";
+import {
+	Button, Alert,
+	Table, TableBody,
+	TableCell, TableRow,
+	TextField, Typography,
+} from "@mui/material";
+import Box from "@mui/material/Box";
+import Container from "@mui/material/Container";
+import Collapse from "@mui/material/Collapse";
+
+function InputFields({ defaultTitle, defaultContent, titleComponenId, contentComponenId }) {
+	const useStyles = makeStyles(() => ({
+		input: {
+			color: "#FFF",
+		},
+	}));
+
+	const classes = useStyles();
+	return <Grid container spacing={2}>
+		<Grid item xs={12}>
+			<TextField
+				required
+				fullWidth
+				name="title"
+				label="Title"
+				id={titleComponenId}
+				color="warning"
+				defaultValue={defaultTitle}
+				inputProps={{ className: classes.input }}
+			/>
+		</Grid>
+		<Grid item xs={12}>
+			<TextField
+				required
+				fullWidth
+				name="content"
+				label="Content"
+				id={contentComponenId}
+				defaultValue={defaultContent}
+				inputProps={{ className: classes.input }}
+			/>
+		</Grid>
+	</Grid>;
+}
+
+function AnnouncementEntry({ announcement, editing, setEditing, userIsAdmin }) {
+	const created_at = new Date(announcement.created_at * 1000);
+	const formated_created_at =
+		created_at.getDate() + "/" +
+		created_at.getMonth() + "/" +
+		created_at.getFullYear() + " " +
+		created_at.getHours() + ":" +
+		created_at.getMinutes();
+
+	const handleEditButton = index => () => {
+		console.log("edit pressed for index " + index);
+		setEditing(index);
+	};
+
+	const handleDeleteButton = index => () => {
+		console.log("Delete pressed for index " + index);
+	};
+
+	const handleCancelButton = () => {
+		setEditing(-1);
+	};
+
+	const handleSubmitButton = (announcementId, titleComponenId, contentComponenId) => () => {
+		const title = document.getElementById(titleComponenId);
+		const content = document.getElementById(contentComponenId);
+		if (title.value.length == 0 || content.value.length == 0) {
+			console.log("missing input or content!");
+			return;
+		}
+		appwriteApi.updateAnnouncement({
+			"title": title.value,
+			"content": content.value
+		}, announcementId)
+			.then(() => {
+				// TODO: update edited announcement
+			})
+			.catch((e) => {
+				console.log(e);
+			});
+		// TODO: update list after new announcement added
+	};
+
+	return <div style={{ width: "100%" }}>
+		<Box sx={{ display: "flex", p: 1, bgcolor: "blue" }}>
+			<Container sx={{ flex: "850%" }}>
+				<Typography variant="h5">{announcement.title}</Typography >
+				<Typography><i>{formated_created_at}</i></Typography >
+				<Typography>{announcement.content}</Typography >
```

Comment:
This is somewhat flawed. An announcement does not consist of a single Paragraph, but multiple. As per acceptance criteria this passes, but we need to change this at some point for sure. We also need to add support for highlighted text (bold, italic etc).